src/algebraic_geometry/prime_spectrum/basic.lean

```diff
@@ -585,6 +585,70 @@ begin
     exact is_localization.comap_map_of_is_prime_disjoint M S _ x.2 h }
 end
 
+section spec_of_surjective
+/-! The comap of a surjective ring homomorphism is a closed embedding between the prime spectra. -/
+
+open function ring_hom
+
+lemma comap_inducing_of_surjective (hf : surjective f) : inducing (comap f) :=
+{ induced := begin
+    simp_rw [topological_space_eq_iff, ←is_closed_compl_iff, is_closed_induced_iff,
+      is_closed_iff_zero_locus],
+    refine λ s, ⟨λ ⟨F, hF⟩, ⟨zero_locus (f ⁻¹' F), ⟨f ⁻¹' F, rfl⟩,
+      by rw [preimage_comap_zero_locus, surjective.image_preimage hf, hF]⟩, _⟩,
+    rintros ⟨-, ⟨F, rfl⟩, hF⟩,
+    exact ⟨f '' F, hF.symm.trans (preimage_comap_zero_locus f F)⟩,
+  end }
+
+lemma image_comap_eq_comap (hf : surjective f) (I : ideal S) :
+  comap f '' zero_locus I = zero_locus (I.comap f) :=
+begin
+  ext p, simp only [set.mem_image, mem_zero_locus, set_like.coe_subset_coe],
+  split,
+  { rintro ⟨p, hp, rfl⟩ a ha, exact hp ha, },
+  { intro h_I_p,
+    have hp : ker f ≤ p.as_ideal :=
+      (ideal.comap_mono bot_le).trans h_I_p,
+    use ideal.map f p.as_ideal,
+    exact ideal.map_is_prime_of_surjective hf hp,
```

Comment:
This suggests you have multiple goals open. Please use another level of `{ .. }` to structure the proof. That will help maintaining the proof in the future.